OmegaExamples/src/probmods/generative-models.jl

```diff
@@ -0,0 +1,612 @@
+### A Pluto.jl notebook ###
+# v0.18.0
+
+using Markdown
+using InteractiveUtils
+
+# ╔═╡ ec0d1440-51ea-11ec-0aef-1f9c2da69ffd
+begin
+    import Pkg
+    # activate the shared project environment
+    Pkg.activate(Base.current_project())
+    using Omega, Distributions, UnicodePlots, OmegaExamples
+end
+
+# ╔═╡ ce896412-97f8-4102-9ed5-bc74dfdb26d4
+md"### Models, simulation, and degrees of belief"
+
+# ╔═╡ cd268cc7-e246-4254-94eb-4809e96b224b
+md"One view of knowledge is that the mind maintains working models of parts of the world. ‘Model’ in the sense that it captures some of the structure in the world, but not all (and what it captures need not be exactly what is in the world—just what is useful). ‘Working’ in the sense that it can be used to simulate this part of the world, imagining what will follow from different initial conditions."
+
+# ╔═╡ f68ca668-e238-4aa2-82be-c2320336e2e9
+md"# Building Generative Models"
+
+# ╔═╡ 2cd7584f-5c78-4d3a-93dc-16da97146e98
+md"We wish to describe in formal terms how to generate states of the world. That is, we wish to describe the causal process, or steps that unfold, leading to some potentially observable states. The key idea of this section is that these generative processes can be described as computations—computations that involve random choices to capture uncertainty about the process.
+Programming languages are formal systems for describing what (deterministic) computation a computer should do. Modern programming languages offer a wide variety of different ways to describe computation; each makes some processes simple to describe and others more complex. However, a key tenet of computer science is that all of these languages have the same fundamental power: any computation that can be described with one programming language can be described by another. (More technically this Church-Turing thesis posits that many specific computational systems capture the set of all effectively computable procedures. These are called universal systems.)"
+
+# ╔═╡ cb28a5c4-a721-4a53-a179-adedf4f71237
+md"""
+[Omega](https://github.com/zenna/Omega.jl) is a programming language that describes probabilistic computation. 
+The key idea is that in Omega you can express stochastic processes and not just deterministic functions (like _and_).
+"""
+
+# ╔═╡ 0ce18f90-f2ef-490a-a7c4-40a6ccfa214a
+md"""
+A basic object in Omega is a random variable class, from which you can generate random variables. For instance, a random variable that is of Bernoulli distribution is generated by -
+"""
+
+# ╔═╡ 354e3b5a-4d1e-43ee-b1bb-e8620fba49e7
+a = 0 ~ Bernoulli()
```

Comment:
I think `a` here represents a random variable (object with type `Member{Bernoulli{Float64}, Int64}`).  Two comments (maybe more for @zenna)

- Perhaps a good convention would be to use capitals or camel case with the object so we remember its a random variable and not one of the RV's outcome?
- The use of "~" immediately calls into mind sampling an outcome.  Then you see the "0" and it's more confusing.  My initial though was that this is some sort of predicate for "0 is sampled from a Bernoulli."  Playing around, I see that "0" is perhaps some sort of name, because I can replace 0 with strings.  Is this 0 like `pyro.sample("0", Bernoulli())`
- Perhaps instead of "plate", this is a "causal Markov kernel", i.e. an expression representing a random variable conditioned on its direct causes?  Just a stab in the dark.
-  But then, below @archanarw says this expression represents a "plate" in a Bayes Net.  Now I'm confused all over.  When I think of plate, I think of a dimension in a model containing multiple conditionally i.i.d. RVs.  The use of 0 seems to imply an index in this dimension, but the size of the dimension itself isn't specified.  Can we contrast this "plate" idea to the plate context manager in Pyro?

I looked at this with a colleague who also has deep experience in PPLs and we're both pretty confused.  I'm not saying the semantics need changing, but perhaps they could be explained better.